### PR TITLE
[Feature]Support crc32 function

### LINKED ISSUE #36256: [FUNCTION] crc32
## crc32

crc32(expr) - Returns a cyclic redundancy check value of the expr as a bigint.

```
select crc32('starrocks');
+--------------------+
| crc32('starrocks') |
+--------------------+
|         2312449062 |
+--------------------+
```

reference:
1. https://trino.io/docs/current/functions/binary.html?highlight=crc32#crc32
2. https://spark.apache.org/docs/3.0.0/api/sql/index.html#crc32
3. https://dev.mysql.com/doc/refman/8.0/en/mathematical-functions.html#function_crc32

### PR DESCRIPTION
**Pull request by leoyy0316**
Why I'm doing:

What I'm doing:

Fixes #36256

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: gensrc/script/functions.py [hunk 1]
```diff
@@ -326,6 +326,7 @@
 
     [30430, 'translate', 'VARCHAR', ['VARCHAR', 'VARCHAR', 'VARCHAR'], 'StringFunctions::translate',
      'StringFunctions::translate_prepare', 'StringFunctions::translate_close'],
+    [30431, 'crc32', 'INT', ['VARCHAR'], 'StringFunctions::crc32'],
```
Comment (fzhedu) [inline gensrc/script/functions.py:329]: change INT to BIGINT?

File: be/src/exprs/string_functions.cpp [hunk 2]
```diff
@@ -4173,5 +4173,12 @@ StatusOr<ColumnPtr> StringFunctions::url_extract_parameter(starrocks::FunctionCo
         return url_extract_parameter_general(columns);
     }
 }
+// crc32
+DEFINE_UNARY_FN_WITH_IMPL(crc32Impl, str) {
+    return static_cast<uint32_t>(crc32_z(0L, (const unsigned char*)str.data, str.size));
```
Comment (LiShuMing) [inline be/src/exprs/string_functions.cpp:4178]: prefer use `HashUtil::zlib_crc_hash`?

File: be/src/exprs/string_functions.cpp [hunk 3]
```diff
@@ -4173,5 +4173,12 @@ StatusOr<ColumnPtr> StringFunctions::url_extract_parameter(starrocks::FunctionCo
         return url_extract_parameter_general(columns);
     }
 }
+// crc32
+DEFINE_UNARY_FN_WITH_IMPL(crc32Impl, str) {
```
Comment (imay) [inline be/src/exprs/string_functions.cpp:4177]: please add sql-tests for this new functions and make sure the crc32 will return the same result with other databases

File: docs/en/sql-reference/sql-functions/function-list.md [hunk 4]
```diff
@@ -330,6 +330,7 @@ You can find a function based on the following categories:
 - [upper](./string-functions/upper.md)
 - [url_decode](./string-functions/url_decode.md)
 - [url_encode](./string-functions/url_encode.md)
+- [crc32](./string-functions/crc32.md)
```
Comment (fzhedu) [inline docs/en/sql-reference/sql-functions/function-list.md:333]: reorder it accordering lex...

File: docs/en/sql-reference/sql-functions/string-functions/crc32.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+# crc32
+
+## Description
+
+This function returns the CRC32 checksum of a string
+
+## Syntax
+
+```Haskell
+BIGINT crc32(VARCHAR str)
```
[the remark this answers is not in the document]
Comment (leoyy0316) [reply docs/en/sql-reference/sql-functions/string-functions/crc32.md:14]: > what if input null?

will return null

File: docs/en/sql-reference/sql-functions/function-list.md (hunk 4 above)
Comment (leoyy0316) [reply docs/en/sql-reference/sql-functions/function-list.md]: > reorder it accordering lex...

done

**Review by fzhedu — APPROVED** [review]

**Review by LiShuMing — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Pull request merged**